### PR TITLE
feat: uses a constant fps clock and updates packages

### PR DESCRIPTION
**Pull request by Eein**
**Introduces a game-engine-like constant time fps clock and cleans up the yolo timer.**

This timer works at 100fps and halts actions in between frames similar to a game loop.
This may prove to be a problem down the road, but worth testing at least.

It also gives us a benchmark to keep the loop under 10ms which we're around 1.5ms at the moment.
I believe it will slow down if the entire frame is filled.

Also updates some packages again. Seems to be fine.

note: puffin isn't up to speed with egui, but they're testing - can restore that back to crates at a later date.

### CODE REVIEW

File: src/state.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 use super::memory::MemoryManagers;
 use crate::game_manager::GameManager;
 use crate::gui::Gui;
+// use puffin_egui::puffin;
```
Comment (orkaboy) [inline src/state.rs:4]: Bit confused about the `puffin` changes overall, since all of it seems to be commented out.

File: src/state.rs [hunk 2]
```diff
@@ -100,9 +96,9 @@ impl State {
     pub fn register_process(&mut self) {
         let process_name = "SeaOfStars.exe";
         // Find the Process
-        match Process::with_name(process_name, &mut self.process_list) {
-            Ok(process) => {
-                if self.context.process.is_none() {
+        if self.context.process.is_none() {
```
Comment (orkaboy) [inline src/state.rs:106]: These changes are unrelated to the fps clock, right?

**Review by orkaboy — APPROVED** [review]

File: src/state.rs (hunk 2 above)
Comment (Eein) [reply src/state.rs:106]: yes - this just reduces the amount of un-needed checks - caught while messing with puffin stuff

File: src/state.rs (hunk 1 above)
Comment (Eein) [reply src/state.rs:4]: I made some adjustments prior to fixing the fps thing, but included them here anyways. 
I still need to set up a custom profile for puffin so you'd do `cargo run --puffin` 

You can ignore these as a TODO - already have a tracking issue for this #37
